tpr/tpr.go

```diff
@@ -0,0 +1,137 @@
+package tpr
+
+import (
+	"fmt"
+	"time"
+
+	microerror "github.com/giantswarm/microkit/error"
+	"github.com/giantswarm/operatorkit/util"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/pkg/api/errors"
+	"k8s.io/client-go/pkg/api/v1"
+	"k8s.io/client-go/pkg/apis/extensions/v1beta1"
+	"k8s.io/client-go/rest"
+)
+
+const (
+	tprInitRetries    = 30
+	tprInitRetryDelay = 3 * time.Second
+)
+
+type Config struct {
+	// Dependencies.
+	Clientset kubernetes.Interface
+	Rest      rest.Interface
+
+	// Settings.
+
+	// Name of the kind of ThirdPartyObjects. It should be in lower case
+	// and hyphen delimited.  Kind names will be converted to CamelCase
+	// when creating ThirdPartyObjects. Hyphens in the kind are assumed to
+	// be word breaks. For instance the kind camel-case would be converted
+	// to CamelCase but camelcase would be converted to Camelcase.
+	Name string
+
+	// Domain of ThirdPartyResource, e.g. example.com. Along with Name must
+	// create an unique pair.
+	Domain string
+
+	// API version, e.g. v1. When creating ThirdPartyObjects will be
+	// prefixed with Domain, e.g. example.com/v1.
+	Version string
+
+	// A free text description
```

Comment:
I would like to start the comment with the property name. This also applies to the comment for `Version` above. In this case here the comment should start with `Description`. Comments should also be full sentences. The full stop at the end is missing.